lib/incoming_mail/handler.rb

```diff
@@ -63,14 +63,17 @@ def create_comment(msg)
       proposal = find_proposal(find_public_id(msg)) or return
       comment_text = find_comment_text(msg)
       comment_user = find_comment_user(msg)
+      comment = nil
       if proposal.has_subscriber?(comment_user)
         # already in the loop, just add comment.
-        proposal.comments.create(comment_text: comment_text, user: comment_user)
+        comment = proposal.comments.create(comment_text: comment_text, user: comment_user)
       else
         # yes, user adds self as observer, which also generates comment
         proposal.add_observer(comment_user, comment_user, comment_text)
-        proposal.comments.last
+        comment = proposal.comments.last
       end
+      Dispatcher.on_comment_created(comment) # sends email
```

Comment:
I think I understand the source of my confusion - this `create_comment` method is doing quite a bit more than creating a comment. It finds a proposal, finds comment text, finds comment user, subscribes the user if they are not already a subscriber, and emails out about the comment. 

It feels like we might want to have a little service class for this. Having a single private method handle all of that means it can be hard (for me, at least!) to understand what's going on and hard to have good unit tests, which would ensure that the behavior doesn't change without us realizing it.

Do you want me to take a stab at a service class for this? I am happy to do it. Also happy for you to do it, whatever makes sense.